rest/src/main/scala/io/udash/rest/raw/RestResponse.scala

```diff
@@ -0,0 +1,95 @@
+package io.udash
+package rest.raw
+
+import com.avsystem.commons._
+import com.avsystem.commons.misc.ImplicitNotFound
+import com.avsystem.commons.rpc.{AsRaw, AsReal}
+
+import scala.annotation.implicitNotFound
+import scala.util.Failure
+
+case class RestResponse(code: Int, headers: IMapping[PlainValue], body: HttpBody) {
+  def toHttpError: HttpErrorException =
+    HttpErrorException(code, body.textualContentOpt.toOptArg)
+  def ensureNonError: RestResponse =
+    if (code >= 200 && code < 300) this else throw toHttpError
+}
+
+object RestResponse extends RestResponseLowPrio {
+  def plain(status: Int, message: OptArg[String] = OptArg.Empty): RestResponse =
+    RestResponse(status, IMapping.empty, HttpBody.plain(message))
+
+  class LazyOps(private val resp: () => RestResponse) extends AnyVal {
+    def recoverHttpError: RestResponse = try resp() catch {
+      case e: HttpErrorException => e.toResponse
+    }
+  }
+  implicit def lazyOps(resp: => RestResponse): LazyOps = new LazyOps(() => resp)
+
+  implicit class AsyncOps(private val asyncResp: RawRest.Async[RestResponse]) extends AnyVal {
```

Comment:
Stay consistent with `LazyOps` in using or not using implicit classes.

Reply:
No, `LazyOps` is not `implicit` only because value classes can't wrap by-name params.